### ISSUE TITLE
Consider a storage option for self-hosted users (or paid subscribers for instance) ?

### ISSUE THREAD

**Issue by Limezy** [buckets/relay#13]
I understand that this is a pure relay, and that no data is stored.
It's a nice feature, since it's probably very cheap to run for thousands of users.
However, it also means that data will only sync when two devices have Buckets opened.
For self-hosted users, both the storage cost and the potential threat of having the buckets data stored in the relay is much less of a concern. Could you please consider an option for which the relay would store the buckets ?
This will be very beneficial as an asynchronous sync will become possible (my wife and I will probably never have Buckets opened at the exact same minute)
Thanks !

**Comment by iffy** [comment]
My plan for users who want to use a service that stores budgets is to let them use whatever file-syncing service they want (Google Drive, Dropbox, WebDAV, iCloud, Syncthing, etc...). When merging is finished, Buckets will grow a "File Server" sharing avenue that expects to have a folder somewhere that it can save files to and read files from.

**Comment by Limezy** [comment]
Your plan makes sense. Do I understand well that, for instance, the iOS app will have an sharing avenue that will interact with the Files app, and then it's your call to use Google Drive, Dropbox, WebDAV, iCloud, Syncthing, Nextcloud or whatever other file syncing service compatible with that same Files app ?

**Comment by iffy** [comment]
@Limezy Yes, that's the intent, though I haven't drilled down into how specifically I'll do that with iOS and Android. It might be that Buckets will have to learn how to interact with individual services, but I'm hoping I can get a simple "here's a folder you can save/read budgets to" interface.
